modules/page_type/default/templates/default.page_type.js.tmpl

```diff
@@ -76,23 +113,65 @@
 					method.call(_this, event, jQuery(this).closest('li'));
 				};
 			};
-			filled_modules.on('click', 'li .remove_button', bind(this.remove_object));
-			filled_modules.on('click', 'li .edit_condition_button', bind(this.edit_condition));
-			filled_modules.on('click', 'li .edit_button', bind(this.edit_object));
-			filled_modules.on('click', 'li .draft_status', bind(this.drop_draft));
-
-			var available_modules = container.find('.available_modules');
-			var delete_drop_zone = container.find('.delete_drop_zone').droppable({
-				accept: 'li[data-container]',
-				tolerance: 'pointer',
-				activeClass: 'ui-state-draggable',
-				hoverClass: 'ui-state-droppable',
-				drop: function(event, ui) {
-					_this.remove_object(event, ui.draggable, true);
-					event.stopPropagation();
+
+			var delete_menu = jQuery('<ul/>');
+			delete_menu.append('<li class="language"><div>{{writeString=wns.language_object.delete.language}}</div></li>');
+			delete_menu.append('<li class="draft"><div>{{writeString=wns.language_object.delete.draft}}</div></li>');
+			delete_menu.append('<li class="all"><div>{{writeString=wns.language_object.delete.all}}</div></li>');
+			var open_delete_menu = add_menu(delete_menu, container, this.container, function selected(event, ui, current_item) {
+				var current_delete_item = current_item.closest('li');
+				if(ui.item.is('.draft')) {
+					return _this.drop_draft(event, current_delete_item);
 				}
+				if(ui.item.is('.language')) {
+					return _this.remove_object(event, current_delete_item, false);
+				}
+				if(ui.item.is('.all')) {
+					return _this.remove_object(event, current_delete_item, true);
+				}
+			}, function opening(current_item) {
+				var current_delete_item = current_item.closest('li');
+				var language_id = current_delete_item.data('languageId');
+				var md = _this.find_structure(current_delete_item);
+				delete_menu.find('li.language').toggle(md.content.language_objects[language_id].exists_in_language);
+				delete_menu.find('li.draft').toggle(md.content.language_objects[language_id].is_draft);
 			});
+			filled_modules.on('click', 'li .remove_button', open_delete_menu);
+			filled_modules.on('click', 'li .edit_condition_button', bind(this.edit_condition));
+			filled_modules.on('click', 'li .edit_button', bind(this.edit_object));
 
+			var available_modules = jQuery('<ul/>');
+			Object.keys(this.possible_frontend_modules).forEach(function(module_name) {
+				var entry = jQuery('<li><div></div></li>').data({moduleName: module_name});
+				entry.find('div').text(_this.possible_frontend_modules[module_name]);
+				available_modules.append(entry);
+			});
+			var open_available_modules = add_menu(available_modules, container, this.container, function selected(event, ui, current_item) {
+				var current_container = current_item.nextAll('.template-container:eq(0)');
+				var current_items = current_container.children();
+				var position = current_items.length;
```

Comment:
Maybe we could make adding the module on the top an option using

```javascript
var position = event.altKey ? 0 : current_items.length;
```

so that if the user holds down option when selecting the module, it will be added on top